Domain/ProductManagement/Inventory.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace LenaInventoryManagementSystem;
+
+public class Inventory
+{
+    private static List<Product> _products = new();
+    private static InventoryUI _inventoryUI=new();
+    private const int _maxItemsInStock = 150;
+
+    internal static void MainMenu()
+    {
+        _inventoryUI.DisplayMainMenu();
+        var userSelection=_inventoryUI.UserSelection();
+
+        switch (userSelection)
+        {
+             case "1":
+                _inventoryUI.AddingANewProductUI();
+                break;
+            case "2":
+                _inventoryUI.DisplayAllProducts();
+                break;
+            case "3":
+                _inventoryUI.UpdateProductUI();
+                break;
+            case "4":
+                _inventoryUI.DeletingAProductUI();
+                break;
+            case "5":
+                _inventoryUI.ViewAProductUI();
+                break;
+            default:
+                _inventoryUI.PrintMessage("Invalid selection. Please try again.");
+                break;
+        }
+    }
+
+    internal static void AddANewProduct(string name, double price, int quantity)
+    {
+        //Checking if the product exists.
+        Product? existingProduct = _products.FirstOrDefault(p => p.ProductName == name);
```

Comment:
I see you have retrieved this but didn't use it. Do you need to check if it's empty or not in the next if statement?

Reply:
if your goal to just check if the product exist, it would be faster to use `.Any` instead of `.FirstOrDefault`, since you don't care about actually retrieving the whole object.